File: sdks/java/src/main/java/com/echoagent/sdk/JsonRpcValues.java
package com.echoagent.sdk;

/** MCP JSON-RPC request/notification values without owning transport. */
public final class JsonRpcValues {
    private JsonRpcValues() {}

    public record Request(String jsonrpc, Object id, String method, Object params) {
        public static Request newRequest(String method, Object params) {
            return new Request("2.0", null, method, params);
        }
    }

    public record Notification(String jsonrpc, String method, Object params) {
        public static Notification newNotification(String method, Object params) {
            return new Notification("2.0", method, params);
        }
    }
}

File: sdks/typescript/src/jsonrpc_values.ts
/** MCP JSON-RPC request/notification values without owning transport. */
export interface JsonRpcRequest {
  readonly jsonrpc: "2.0";
  readonly id: unknown | null;
  readonly method: string;
  readonly params: unknown | null;
}

export interface JsonRpcNotification {
  readonly jsonrpc: "2.0";
  readonly method: string;
  readonly params: unknown | null;
}

export function jsonRpcRequest(method: string, params: unknown | null = null): JsonRpcRequest {
  return Object.freeze({ jsonrpc: "2.0" as const, id: null, method, params });
}

export function jsonRpcNotification(method: string, params: unknown | null = null): JsonRpcNotification {
  return Object.freeze({ jsonrpc: "2.0" as const, method, params });
}
